ext/parquet/decoder.go

```diff
@@ -45,10 +45,12 @@ func (w *decoder) Decode(v interface{}) error {
 }
 
 func (w *decoder) Close() (err error) {
-	for _, c := range w.closers {
-		if e := c.Close(); e != nil {
-			err = e
-		}
+	// close the tmp file if present
+	if w.tmp != nil {
+		err = multierr.Append(err, w.tmp.Close())
```

Comment:
you also need to is.Remove the file too

Reply:
Your original tmpfile struct does that as part of it's `Close()`